core/src/main/java/org/bitcoinj/crypto/DeterministicKey.java

```diff
@@ -91,6 +97,64 @@ public DeterministicKey(ImmutableList<ChildNumber> childNumberPath,
         this.keyCrypter = checkNotNull(crypter);
     }
 
+    /**
+     * Return the fingerprint of this key's parent as an int value, or zero if this key is the
+     * root node of the key hierarchy.  Raise an exception if the arguments are inconsistent.
+     * This method exists to avoid code repetition in the constructors.
+     */
+    private int ascertainParentFingerprint(DeterministicKey parentKey, byte[] parentFingerprint)
```

Comment:
> Shouldn't parentFingerprint become an int everywhere?

Not unless we're going to change the return value of [getFingerprint()](https://github.com/bitcoinj/bitcoinj/blob/master/core/src/main/java/org/bitcoinj/crypto/DeterministicKey.java#L143) to `int` also.  The key fingerprints returned by different methods of a single class ought to be represented as a consistent type, especially since one purpose of having the fingerprint of a key's parent is to identify that parent and the probable way to do that is to compare the return values of one key's `getParentFingerprint()` with another's `getFingerprint()`.  It seems wrong that a client in that expectable situation would have to convert the return type of one of those functions just to see if they represent the same fingerprint.  I don't have a strong preference as to which that type should be, but those two methods ought to be consistent with each other, don't you think?

Are you referring as well to the second argument to `ascertainParentFingerprint`?  That byte array already exists before this patch, being set on [line 397](https://github.com/bitcoinj/bitcoinj/blob/master/core/src/main/java/org/bitcoinj/crypto/DeterministicKey.java#L397).  We can pass that argument as an `int`, but that means either initializing an `int` just to copy its value from another variable that's already initialized, or else deserializing it from the `ByteBuffer` as an `int`.  If you want, I'll do that; I was trying to modify existing code as little as possible.